signing/src/main/java/tech/pegasys/web3signer/signing/BlsKeystoreBulkLoader.java

```diff
@@ -39,36 +39,59 @@
 public class BlsKeystoreBulkLoader {
   private static final Logger LOG = LogManager.getLogger();
 
-  public Collection<ArtifactSigner> loadKeystoresUsingPasswordDir(
+  public SecretValueResult<ArtifactSigner> loadKeystoresUsingPasswordDir(
       final Path keystoresDirectory, final Path passwordsDirectory) {
-    final List<Path> keystoreFiles = keystoreFiles(keystoresDirectory);
+    final List<Path> keystoreFiles;
+    try {
+      keystoreFiles = keystoreFiles(keystoresDirectory);
+    } catch (final IOException e) {
+      LOG.error("Error reading keystore files", e);
+      return SecretValueResult.errorResult();
+    }
+
     return keystoreFiles.parallelStream()
         .map(
             keystoreFile ->
                 createSignerForKeystore(
                     keystoreFile,
                     key -> Files.readString(passwordsDirectory.resolve(key + ".txt"))))
-        .flatMap(Optional::stream)
-        .collect(Collectors.toList());
+        .reduce(
+            SecretValueResult.newInstance(new HashSet<>(), 0),
+            (r1, r2) -> {
+              r1.merge(r2);
```

Comment:
nit: if this was a static function that returned a SecretValueResult then you use a method reference here